src/app-layer-expectation.c

```diff
@@ -85,6 +89,12 @@ typedef struct ExpectationData_ {
     void (*DFree)(void *);
 } ExpectationData;
 
+typedef struct ExpectationList_ {
+    Expectation *cqh_first; /* first element  */
```

Comment:
shouldn't `CIRCLEQ_HEAD` be used here?

Reply:
We have
```
#define CIRCLEQ_HEAD(name, type)					\
struct name {								\
	struct type *cqh_first;		/* first element */		\
	struct type *cqh_last;		/* last element */		\
}
```
And I wanted to add the length variable to the structure, so to avoid to add one layer in the structure, I did decide to directly use the directly fields in the ExpectationList structure.

Reply:
Whats the reason for avoiding this extra layer? I don't think it matters at all after compilation. Seem just cleaner to me to use the api as intended.